textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -36,5 +50,27 @@ public boolean satisfy(ITuple tuple) {
         }
         return false;
     }
+    
+    public List<Span> statisfySpan(ITuple tuple) {
+    	List<Span> res = new ArrayList<>();
+    	if (tuple == null) {
+    		return res; //empty array
+    	}
+    	IField field = tuple.getField(fieldName);

```

Comment:
@sandeepreddy602 @inkudo @rajesh9625 : per our discussion on Friday, should we support both StringField and TextField here for regex matching?